spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/segmentation/COCODataset.scala

```diff
@@ -0,0 +1,348 @@
+  @SerializedName("date_captured") var dateCaptured: String = _
+  @SerializedName("file_name") var fileName: String = _
+
+  def dumpTo(context: COCOSerializeContext, dataset: COCODataset): Unit = {
+    context.dump(height)
+    context.dump(width)
+    context.dump(annotations.size)
+    annotations.foreach(_.dumpTo(context, dataset))
+  }
+
+  /**
+   * Get the path of the image in local file system
+   * @return
+   */
+  def path: Path = Paths.get(imgRootPath, fileName)
+
+  /**
+   * Read the data from the image file
+   * @return
+   */
+  def data: Array[Byte] = Files.readAllBytes(path)
+
+}
+
+/**
+ * An annotation for an image (OD in the name for Object Detection)
+ * @param id
+ * @param imageId the Id of the image
+ * @param categoryId the Id of the category. Note that categoryId is not continuous from 0 to
+ *                   the number of categories. You can use COCODataset.cateId2Idx to convert an
+ *                   categoryId to a compact category index.
+ * @param segmentation the segmentation data
+ * @param area  area
+ * @param bbox  the bounding box, (xmin, ymin, xmax, ymax)
+ * @param isCrowd if the annotation is a crowd. e.g. a crowd of people. If true, segmentation is
+ *                an COCORLE object
+ * @param image the reference to the image
+ */
+case class COCOAnotationOD(id: Long, imageId: Long, categoryId: Long,
+  segmentation: COCOSegmentation, area: Float, bbox: (Float, Float, Float, Float), isCrowd: Boolean,
+  @transient var image: COCOImage = null
+) {
+  def dumpTo(context: COCOSerializeContext, dataSet: COCODataset): Unit = {
+    context.dump(dataSet.categoryId2Idx(categoryId))
+    context.dump(area)
+    context.dump(bbox._1)
+    context.dump(bbox._2)
+    context.dump(bbox._3)
+    context.dump(bbox._4)
+    context.dump(isCrowd)
+    segmentation.dumpTo(context)
+  }
+}
+
+case class COCOLicence(
+  id: Long, name: String, url: String
+)
+
+case class COCOCategory(
+  id: Long, name: String) {
+  @SerializedName("supercategory") var superCategory: String = _
+}
+
+trait COCOSegmentation {
+  def dumpTo(context: COCOSerializeContext): Unit
+}
+
+case class COCOPoly(_poly: Array[Array[Float]])
+  extends PolyMasks(_poly, -1, -1) with COCOSegmentation {
+  private[segmentation] var _height: Int = -1
+  private[segmentation] var _width: Int = -1
+  override def dumpTo(context: COCOSerializeContext): Unit = {
+    context.dump(poly.length)
+    poly.foreach(p => {
+      context.dump(p.length)
+      p.foreach(xy => {
+        context.dump(xy)
+      })
+    })
+  }
+  override def height: Int = _height
+  override def width: Int = _width
+}
+
+ case class COCORLE(_counts: Array[Int], _height: Int, _width: Int)
+   extends RLEMasks(_counts, _height, _width) with COCOSegmentation {
+   override def dumpTo(context: COCOSerializeContext): Unit = {
+     context.dump(counts.length)
+     counts.foreach(p => {
+       context.dump(p)
+     })
+   }
+ }
+
+object COCODataset {
```

Comment:
So how to get whole COCO dataset from this object?

Reply:
Can now load data by COCOImage